### PR TITLE
ENH: add ioc fields to EnvironmentalMonitor happi container

### PR DESCRIPTION
**Pull request by sainyamn**
[ECS-4106](https://jira.slac.stanford.edu/browse/ECS-4106)
#################

This feature is used to modify card num and channel number in EnvironmentalMonitor, instead of previously fixed values of 1.

Motivation and Context
--------
- To add fields of ioc_chan_num and ioc_card_num in happi for EnvironmentalMonitor device.
- Used in MODS tile_move script to facilitate automated configuration file generation. 
- In order to automate IOC config file generation it is useful to keep track of ion_card_num and ion_chan_num for the Environmental Sensors, specifically used in EK9000s. 
- By default, the port and chan num are 1 however they may be different in case of more than one env sensors. 

Features
--------
- ioc_chan_num and ion_card_num were added to EnvironmentalMonitor in containers.py 

Testing
--------
-  Procedure outlined in confluence is followed: [https://confluence.slac.stanford.edu/display/PCDS/PCDS+Conda+Python+Environments#PCDSCondaPythonEnvironments-CanItestmychangeslocally,withoutmodifyingtheglobaldev:~:text=the%20highest%20consistency.-,1.%20Testing%20One%20Package%20with%20PYTHONPATH,-This%20is%20a]
`$ cd /reg/g/pcds/epics-dev/nagar123/mods/happi_container_mod/pcdsdevices`
`$ source pcds_conda`
`$ PYTHONPATH=$PWD`
`$ ipython`
In [1]: from happi import Client, HappiItem
   ...: client = Client(path='/local/db.json')
In [2]: from pcdsdevices.happi.containers import EnvironmentalMonitor
In [3]: client.search(name="lm1k4_inj_env_sens")
In [4]: env_mon=client.search(name="lm1k4_inj_env_sens")
In [5]: env_mon=env_mon[0]
In [6]: env_mon['name']
Out[6]: 'lm1k4_inj_env_sens'
In [7]: env_mon['ioc_card_num'] 
returns blank no error, final check:
In [8]:print(dir(EnvironmentalMonitor))
contains the ioc_chan_num and ioc_card_num in the list of fields.

Documentation
------------
- changes documented in upcoming release notes as 'Test br #1215 Add card, chan num in EnvironmentalMonitor'

Contributors
------------
- Sainyamn

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [ ] ~New/changed functions and methods are covered in the test suite where possible~
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

File: docs/source/upcoming_release_notes/1215-Add_ioc_card_and_chan_num_in_EnvironmentalMonitor
Comment (tangkong) [inline docs/source/upcoming_release_notes/1215-Add_ioc_card_and_chan_num_in_EnvironmentalMonitor:1]: You should follow the template provided by the pre-release notes script.  This involves running the script and adding items under the headers provided.  

You can't simply copy the PR description into the file, unfortunately.  There's a specific format our release notes infrastructure expects

File: pcdsdevices/happi/containers.py [hunk 1]
```diff
@@ -666,6 +666,10 @@ class EnvironmentalMonitor(LCLSItem):
                        optional=True, enforce=str)
     ioc_base = EntryInfo('Base PV of the EK9000 IOC', optional=True,
                          enforce=str)
+    ioc_chan_num = EntryInfo('Channel number for the environmental monitor.',
+                            optional=True, enforce=str)
+    ioc_card_num = EntryInfo('Card number for the environmental monitor.',
+                            optional=True, enforce=str)
```
Comment (ZLLentz) [inline pcdsdevices/happi/containers.py:672]: the pre-commit flake8 error comes from the non-matching indentation here:
```suggestion
    ioc_chan_num = EntryInfo('Channel number for the environmental monitor.',
                             optional=True, enforce=str)
    ioc_card_num = EntryInfo('Card number for the environmental monitor.',
                             optional=True, enforce=str)
```

**Comment by ZLLentz** [comment]
fwiw I think the changes look good and the testing was sufficient

File: docs/source/upcoming_release_notes/1215-Add_ioc_card_and_chan_num_in_EnvironmentalMonitor
Comment (ZLLentz) [inline docs/source/upcoming_release_notes/1215-Add_ioc_card_and_chan_num_in_EnvironmentalMonitor:1]: I think we need to delete this older version of the file and then this is good to go

**Review by ZLLentz — APPROVED** [review]
Looks good to me now!

**Pull request merged**